exercises/array-manipulation-2/src/chunk.js

```diff
@@ -0,0 +1,19 @@
+/* exported chunk */
+function chunk(array, size) {
+  var storageArray = [];
+  var newArray = [];
+  if (array.length === 0) {
+    return newArray;
+  }
+  for (var i = 0; i < array.length; i++) {
+    storageArray.push(array[i]);
+    if (storageArray.length === size) {
+      newArray.push(storageArray);
+      storageArray = [];
+    }
+    if ((i + 1 === array.length) && (storageArray.length > 0)) {
+      newArray.push(storageArray);
+    }
```

Comment:
You can move this conditional to after the loop runs and remove the expression checking `i` and get the same result